libvcx_core/src/api_vcx/api_global/wallet.rs

```diff
@@ -123,16 +126,19 @@ pub async fn wallet_update_wallet_record_value(xtype: &str, id: &str, value: &st
 
 pub async fn wallet_update_wallet_record_tags(xtype: &str, id: &str, tags_json: &str) -> LibvcxResult<()> {
     let wallet = get_main_wallet();
-    map_ariesvcx_core_result(wallet.update_wallet_record_tags(xtype, id, tags_json).await)
+    let tags: HashMap<String, String> = serde_json::from_str(tags_json)?;
+    map_ariesvcx_core_result(wallet.update_wallet_record_tags(xtype, id, tags).await)
 }
 
 pub async fn wallet_add_wallet_record_tags(xtype: &str, id: &str, tags_json: &str) -> LibvcxResult<()> {
     let wallet = get_main_wallet();
-    map_ariesvcx_core_result(wallet.add_wallet_record_tags(xtype, id, tags_json).await)
+    let tags: HashMap<String, String> = serde_json::from_str(tags_json)?;
+    map_ariesvcx_core_result(wallet.add_wallet_record_tags(xtype, id, tags).await)
 }
 
 pub async fn wallet_delete_wallet_record_tags(xtype: &str, id: &str, tags_json: &str) -> LibvcxResult<()> {
     let wallet = get_main_wallet();
+    let tags_json = serde_json::from_str(tags_json)?;
```

Comment:
I believe this is not necessary and may cause bugs (unfortunately I believe this might be untested code, so CI is not reporting issues).

`wallet.delete_wallet_record_tags` still just takes in a regular `&str`. In the future we should fix the typing for `delete_wallet_record_tags`, but that can be a job for later.

Reply:
alright!